From 9bdfc62dd8ddd4d501cc235556f2294933a99dc1 Mon Sep 17 00:00:00 2001
From: Jae Hyun Yoo <jae.hyun.yoo@intel.com>
Date: Tue, 6 Nov 2018 16:25:58 -0800
Subject: [PATCH] mfd: peci: Add ICX generation info

This commit adds CPU generation info for ICX family. It should be
managed within downstream until it is rolled out.

Signed-off-by: Jae Hyun Yoo <jae.hyun.yoo@intel.com>
---
 drivers/mfd/intel-peci-client.c       | 6 ++++++
 include/linux/mfd/intel-peci-client.h | 5 +++++
 2 files changed, 11 insertions(+)

diff --git a/drivers/mfd/intel-peci-client.c b/drivers/mfd/intel-peci-client.c
index d62442438512..e276ac5f05ee 100644
--- a/drivers/mfd/intel-peci-client.c
+++ b/drivers/mfd/intel-peci-client.c
@@ -43,6 +43,12 @@ static const struct cpu_gen_info cpu_gen_info_table[] = {
 		.core_max      = CORE_MAX_ON_SKX,
 		.chan_rank_max = CHAN_RANK_MAX_ON_SKX,
 		.dimm_idx_max  = DIMM_IDX_MAX_ON_SKX },
+	{ /* Icelake Xeon */
+		.family        = 6, /* Family code */
+		.model         = INTEL_FAM6_ICELAKE_X,
+		.core_max      = CORE_MAX_ON_ICX,
+		.chan_rank_max = CHAN_RANK_MAX_ON_ICX,
+		.dimm_idx_max  = DIMM_IDX_MAX_ON_ICX },
 };
 
 static int peci_client_get_cpu_gen_info(struct peci_client_manager *priv)
diff --git a/include/linux/mfd/intel-peci-client.h b/include/linux/mfd/intel-peci-client.h
index dd5eb36cca75..ced1fb268b74 100644
--- a/include/linux/mfd/intel-peci-client.h
+++ b/include/linux/mfd/intel-peci-client.h
@@ -17,6 +17,7 @@
 #define INTEL_FAM6_HASWELL_X   0x3F
 #define INTEL_FAM6_BROADWELL_X 0x4F
 #define INTEL_FAM6_SKYLAKE_X   0x55
+#define INTEL_FAM6_ICELAKE_X   0x6A
 #endif
 
 #define CORE_MAX_ON_HSX        18 /* Max number of cores on Haswell */
@@ -31,6 +32,10 @@
 #define CHAN_RANK_MAX_ON_SKX   6  /* Max number of channel ranks on Skylake */
 #define DIMM_IDX_MAX_ON_SKX    2  /* Max DIMM index per channel on Skylake */
 
+#define CORE_MAX_ON_ICX        28 /* Max number of cores on Icelake */
+#define CHAN_RANK_MAX_ON_ICX   8  /* Max number of channel ranks on Icelake */
+#define DIMM_IDX_MAX_ON_ICX    2  /* Max DIMM index per channel on Icelake */
+
 #define CORE_NUMS_MAX          CORE_MAX_ON_SKX
 #define CHAN_RANK_MAX          CHAN_RANK_MAX_ON_HSX
 #define DIMM_IDX_MAX           DIMM_IDX_MAX_ON_HSX
-- 
2.7.4

